### PR TITLE
[RNMobile] Remove title focus for default editor view

### PR DESCRIPTION
**Pull request by SiobhyB**
## Related PRs

* `Gutenberg Mobile:` https://github.com/wordpress-mobile/gutenberg-mobile/pull/4740
* `iOS:` https://github.com/wordpress-mobile/WordPress-iOS/pull/18338

## What?

By default, when a user first opens the post editor within the WordPress mobile app, the title is focused. This PR changes that and makes it so that there is no default focus when opening the editor.

## Why?

User research indicates that it's not always clear to users what primary actions are available when they first open the editor. By iterating on the default view of the editor, it's hoped that it will be easier for users to discover blocks and, therefore, improve the overall writing/onboarding experience.

This PR is part of a series of changes we're making to the editor's default view within the app. Another, related to change to the way the block inserter button is styled within the default view can be found in https://github.com/WordPress/gutenberg/pull/39726.

## How?

All references to `setFocusOnTitle` have been removed from the Gutenberg and iOS codebase. Note, it wasn't necessary to remove any references from the Android codebase, as all Android-related references exist within Gutenberg. 

## Testing Instructions

* When opening the post editor within the iOS or Android apps, verify that the title is _not_ focused by default.
* Experiment with various writing and publishing flows as part of testing. For example, make sure that the title _isn't_ focused when saving and returning to a post, verify that there are no regressions when going through [the writing flow test cases](https://github.com/wordpress-mobile/test-cases/edit/trunk/test-cases/gutenberg/writing-flow/readme.md), etc. The title field should only be focused when adding text.

### Possible Regressions 

Possible regressions involve the title not being focused during times it should, so it'd be especially helpful to verify that the title remains in focus during key writing flows, such as switching between portrait/landscape while typing a title.

## Screenshots or screencast 

| Before (Title Focus)  | After (No Focus) |
| ------------- | ------------- |
|  |  |

### CODE REVIEW

**Review by iamthomasbishop — APPROVED** [review]
Seems to be working as expected from a design perspective. Thanks @SiobhyB!

File: packages/edit-post/src/editor.native.js [hunk 1]
```diff
@@ -88,14 +85,6 @@ class Editor extends Component {
 	componentDidMount() {
 		const { editEntityRecord, postType, postId } = this.props;
 
-		this.subscriptionParentSetFocusOnTitle = subscribeSetFocusOnTitle(
-			() => {
-				if ( this.postTitleRef ) {
```
Comment (fluiddot) [inline packages/edit-post/src/editor.native.js:93]: Looks like `postTitleRef` was only being used here, so I'm wondering if we could also remove the references from the following parts:

https://github.com/WordPress/gutenberg/blob/4f3b456bf84bc7e2373898d72028c6877261d003/packages/edit-post/src/editor.native.js#L109-L111

https://github.com/WordPress/gutenberg/blob/4f3b456bf84bc7e2373898d72028c6877261d003/packages/edit-post/src/editor.native.js#L163

File: packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java [hunk 2]
```diff
@@ -39,7 +39,6 @@ public class RNReactNativeGutenbergBridgeModule extends ReactContextBaseJavaModu
     private static final String EVENT_NAME_REQUEST_GET_HTML = "requestGetHtml";
     private static final String EVENT_NAME_UPDATE_HTML = "updateHtml";
     private static final String EVENT_NAME_UPDATE_TITLE = "setTitle";
-    private static final String EVENT_NAME_FOCUS_TITLE = "setFocusOnTitle";
```
Comment (fluiddot) [inline packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java:42]: Similar to this change, we can do the same on iOS and remove the reference to the `setFocusOnTitle` event:

https://github.com/WordPress/gutenberg/blob/4f3b456bf84bc7e2373898d72028c6877261d003/packages/react-native-bridge/ios/RNReactNativeGutenbergBridge.swift#L413

**Comment by hypest** [comment]
👋 sorry for the kinda late notice but, after discussing it a bit in Slack `1649253951.736939-p1649758705434009-slack-C03840H2X0W`, we are not planning to remove the bit for not-focusing the title when a new post. So, a new post should attract the focus to the title input, while editing an already drafted/published/etc post should focus nothing on entering. Please let's update the PR to that end. Thanks!

**Comment by SiobhyB** [comment]
@fluiddot, @hypest, @iamthomasbishop, thank you for reviewing and the heads up! I'll go ahead to close this set of PRs given the title focus already works as described here:

> a new post should attract the focus to the title input, while editing an already drafted/published/etc post should focus nothing on entering.

Please let me know if I'm missing anything and feel welcome to re-open if there is anything we need to iterate on with the title focus.

I think this leads to a discussion around whether we'd like the "expanded" button style to display when an empty post is first loaded (and the title is focused) but that feels worthy of a separate PR/discussion to this.

**Comment by hypest** [comment]
> I think this leads to a discussion around whether we'd like the "expanded" button style to display when an empty post is first loaded (and the title is focused) but that feels worthy of a separate PR/discussion to this.

Yeap, agreed 👍

**Pull request closed without merging**